### ISSUE TITLE
Subpixel rendering on a transparent background causes artifacts

### ISSUE THREAD

**Issue by theOperand** [alacritty/alacritty#4552]
### System

OS: Arch Linux
Version: alacritty 0.6.0 (04cbf767)
X11 KDE with Kwin

### Description

When using subpixel rendering with a transparent window, the rendering uses the opaque background color instead of the transparent background color. This results in ugly artifacts on backgrounds that don't match the background color.

The following screenshots were taken with a completely transparent background and a black background color, on alternatively a black and a white background (scaled up by 4 in each dimension in GIMP):
![lolcatgood](https://user-images.githubusercontent.com/3975713/101842990-9138c400-3b49-11eb-88cc-842aa8943952.png)
![lolcatbad](https://user-images.githubusercontent.com/3975713/101843008-97c73b80-3b49-11eb-9116-e2d86e9d17e9.png)

The effect is especially pronounced with text that closely matches the background color, for example this screenshot taken on a white background, but with 0.65 opacity and KDE's blur option:
![Screenshot_20201211_003928](https://user-images.githubusercontent.com/3975713/101843069-b594a080-3b49-11eb-8427-47fb3916bd06.png)

**Comment by chrisduerr** [comment]
I'm not sure exactly what you mean and why you use colored text to demonstrate it. It should be far easier to make out subpixel rendering issues with white text. If the background color is a problem, it probably also makes sense to set that to some really noticable value like `#ff00ff`.

Are you able to get pink subpixel rendering using that setup?

**Comment by theOperand** [comment]
You are right that it doesn't really seem to change with the background color and I should have done proper tests before giving that hypothesis, sorry. 

However, I hope that this screenshot properly shows the problem. Top is subpixel rendering off, bottom is subpixel rendering on (background color #ff00ff, opacity 0). It adds a strange "shadow" to all the letters.
![Screenshot_20201211_141230](https://user-images.githubusercontent.com/3975713/101907488-eeb92900-3bba-11eb-9aac-c8b5b629bad9.png)

And, as noted, these artifacts are only visible over light backgrounds, over dark backgrounds they look fine:
![Screenshot_20201211_141907](https://user-images.githubusercontent.com/3975713/101908107-da296080-3bbb-11eb-9633-4db0cecb7180.png)

The artifacts are also not visible over equally bright opaque backgrounds:
![image](https://user-images.githubusercontent.com/3975713/101908333-35f3e980-3bbc-11eb-9e95-4856981680cd.png)

**Comment by chrisduerr** [comment]
For cyan text it looks a bit strange, but looking closely at text above a white background it looks to me like it seems to be doing things about right?

![tmp](https://user-images.githubusercontent.com/8886672/101971606-97eb3800-3c2a-11eb-80a9-8633147a0aab.png)

With these higher resolution images you can see relatively well that the fringes of the white and yellow text at the right side definitely seem red-ish. While at least the white text at the left is blue-ish.

Generally the color you have behind it shouldn't matter, since blending the colors at that point is up to your compositor (Alacritty doesn't know anything about the color behind the window). So I'm not sure if Alacritty is actually doing something wrong here.

@kchibisov do you have an idea, does this look right to you?

**Comment by kchibisov** [comment]
>@kchibisov do you have an idea, does this look right to you?

I'm not entirely sure, need to compare to manual blending or something like that. It could be that background should be corrected properly.

**Comment by theOperand** [comment]
Surely it cant be correct if there's such a difference between subpixel
rendering on complete transparency on white and subpixel rendering on an
opaque white background?

**Comment by kchibisov** [comment]
I mean that you should do manual correction for subpixel, etc.

**Comment by chrisduerr** [comment]
> I mean that you should do manual correction for subpixel, etc.

Correct for what exactly? We have no knowledge of white/black behind it.

**Comment by kchibisov** [comment]
We know alpha value though, so we can adjust alpha mask, I can't say for sure that it's not possible to do anything, thats all I'm trying to say.

And there's https://github.com/alacritty/alacritty/issues/2 , so I'd at least try to understand this issue.

**Comment by kchibisov** [comment]
I mean, you control subpixel rendering in a fontconfig, can just disable it.

**Comment by kchibisov** [comment]
I'll look into it once I've done with gles2 renderer.  Could you test https://github.com/alacritty/alacritty/pull/5902 if it changes anything?
